crates/node/src/lib.rs

```diff
@@ -335,6 +336,16 @@ impl Node {
             return Ok(());
         }
 
+        let Some(sender_key) = self.ctx_manager.get_sender_key(&context_id, &author_id)? else {
+            return self.initiate_sync(context_id, source).await;
+        };
+
+        let shared_key = SharedKey::from_sk(&sender_key);
+
+        let artifact = &shared_key
+            .decrypt(artifact, [0; 12])
```

Comment:
thought I'd mention.. ideally, IVs shouldn't be constant as that impairs the effectiveness of GCM encryption and this can be passed in plain text, but that's an easy fix for a later PR, we can merge with this